pkg/ccl/backupccl/backup_test.go

```diff
@@ -6093,66 +6115,44 @@ func getMockTableDesc(
 	return tabledesc.NewBuilder(&mockTableDescriptor).BuildImmutableTable()
 }
 
-// Unit tests for the getLogicallyMergedTableSpans() method.
-// TODO(pbardea): Add ADDING and DROPPING indexes to these tests.
-func TestLogicallyMergedTableSpans(t *testing.T) {
+// Unit tests for the getPublicIndexTableSpans() method.
+func TestPublicIndexTableSpans(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	ctx := context.Background()
 	codec := keys.TODOSQLCodec
 	unusedMap := make(map[tableAndIndex]bool)
 	testCases := []struct {
-		name                       string
-		checkForKVInBoundsOverride func(start, end roachpb.Key, endTime hlc.Timestamp) (bool, error)
-		tableID                    descpb.ID
-		pkIndex                    descpb.IndexDescriptor
-		indexes                    []descpb.IndexDescriptor
-		addingIndexes              []descpb.IndexDescriptor
-		droppingIndexes            []descpb.IndexDescriptor
-		expectedSpans              []string
+		name            string
+		tableID         descpb.ID
+		pkIndex         descpb.IndexDescriptor
+		indexes         []descpb.IndexDescriptor
+		addingIndexes   []descpb.IndexDescriptor
+		droppingIndexes []descpb.IndexDescriptor
+		expectedSpans   []string
 	}{
 		{
-			name: "contiguous-spans",
-			checkForKVInBoundsOverride: func(start, end roachpb.Key, endTime hlc.Timestamp) (bool, error) {
-				return false, nil
-			},
+			name:          "contiguous-spans",
 			tableID:       55,
 			pkIndex:       getMockIndexDesc(1),
 			indexes:       []descpb.IndexDescriptor{getMockIndexDesc(1), getMockIndexDesc(2)},
-			expectedSpans: []string{"/Table/55/{1-3}"},
+			expectedSpans: []string{"/Table/55/{1-2}", "/Table/55/{2-3}"},
```

Comment:
Why don't these get merged?

Reply:
Nevermind, I see that we are testing the function that _doesn't_ merge things here.

Reply:
I wonder if we should take a few of these test cases and also put them around spansForAllTableIndexes just to test the case where we think we should be getting _some_ merging still.

Reply:
This test changed to unit testing the method that used to be called `getLogicallyMergedSpans` that now returns the unmerged active index spans of the table being backed up. The merging happens outside of the scope of this method. 
I could change this test to work similarly to `TestProtectedTimestampSpanSelectionDuringBackup` and capture the post-merge backed up spans, but I'm also not sure how much we gain from testing `roachpb.MergeSpans` that is well tested as a separate component?

Reply:
whoops didn't see your follow ups.

Reply:
added a subtest that calls `spansForAllTableIndexes` as well.